File: backend/app/schemas/sources.py
"""Schemas for the external data-source surface (/api/sources, WMS layers)."""

from typing import Any

from pydantic import BaseModel


class SourceInfo(BaseModel):
    """A consumable external data source."""

    id: str
    name: str
    kind: str
    base_url: str
    formats: list[str]
    auth_required: bool
    notes: str


class DatasetSummary(BaseModel):
    """A condensed CKAN dataset entry."""

    id: str
    title: str
    organization: str | None = None
    formats: list[str] = []
    url: str | None = None


class WmsLayer(BaseModel):
    """A MapLibre-ready WMS raster layer descriptor."""

    id: str
    name: str
    level: str
    type: str = "raster"
    tiles: list[str]
    tileSize: int = 256
    srid: int = 4326


class CandidaturasResponse(BaseModel):
    """Opaque pass-through wrapper for Candidaturas MX payloads."""

    source: str = "candidaturas_mx"
    data: Any

File: frontend/src/types/sources.ts
export interface SourceInfo {
  id: string;
  name: string;
  kind: "api" | "wms" | "download" | "portal";
  base_url: string;
  formats: string[];
  auth_required: boolean;
  notes: string;
}

export interface DatasetSummary {
  id: string;
  title: string;
  organization: string | null;
  formats: string[];
  url: string | null;
}

export interface WmsLayer {
  id: string;
  name: string;
  level: string;
  type: string;
  tiles: string[];
  tileSize: number;
  srid: number;
}

export interface WmsLayersResponse {
  configured: boolean;
  layers: WmsLayer[];
}
